internal/inventory/gcpfetcher/fetcher_assets.go

```diff
@@ -103,3 +108,74 @@ func (f *assetsInventory) fetch(ctx context.Context, assetChan chan<- inventory.
 		)
 	}
 }
+
+func (f *assetsInventory) findRelatedAssetIds(subType inventory.AssetSubType, item *gcpinventory.ExtendedGcpAsset) []string {
+	ids := []string{}
+	ids = append(ids, item.Ancestors...)
+	if item.Resource != nil {
+		ids = append(ids, item.Resource.Parent)
+	}
+
+	ids = append(ids, f.findRelatedAssetIdsForSubType(subType, item)...)
+
+	ids = lo.Compact(ids)
+	ids = lo.Uniq(ids)
+	return ids
+}
+
+func (f *assetsInventory) findRelatedAssetIdsForSubType(subType inventory.AssetSubType, item *gcpinventory.ExtendedGcpAsset) []string {
+	ids := []string{}
+
+	var fields map[string]*structpb.Value
+	if item.Resource != nil && item.Resource.Data != nil {
+		fields = item.GetResource().GetData().GetFields()
+	}
+
+	switch subType {
+	case inventory.SubTypeGcpInstance:
+		if v, ok := fields["networkInterfaces"]; ok {
+			for _, networkInterface := range v.GetListValue().GetValues() {
+				networkInterfaceFields := networkInterface.GetStructValue().GetFields()
+				ids = appendIfExists(ids, networkInterfaceFields, "network")
+				ids = appendIfExists(ids, networkInterfaceFields, "subnetwork")
+			}
+		}
+		if v, ok := fields["serviceAccounts"]; ok {
+			for _, serviceAccount := range v.GetListValue().GetValues() {
+				serviceAccountFields := serviceAccount.GetStructValue().GetFields()
+				ids = appendIfExists(ids, serviceAccountFields, "email")
+			}
+		}
+		if v, ok := fields["disks"]; ok {
+			for _, disk := range v.GetListValue().GetValues() {
+				diskFields := disk.GetStructValue().GetFields()
+				ids = appendIfExists(ids, diskFields, "source")
+			}
+		}
+		ids = appendIfExists(ids, fields, "machineType")
+		ids = appendIfExists(ids, fields, "zone")
+	case inventory.SubTypeGcpFirewall, inventory.SubTypeGcpSubnet:
+		ids = appendIfExists(ids, fields, "network")
+	case inventory.SubTypeGcpProject, inventory.SubTypeGcpBucket:
+		if item.IamPolicy == nil {
+			break
+		}
+		for _, binding := range item.IamPolicy.Bindings {
+			ids = append(ids, binding.Role)
+			ids = append(ids, binding.Members...)
+		}
+	default:
+		f.logger.Warnf("cannot find related asset IDs for unsupported sub-type %q", subType)
```

Comment:
Do we really need a warn here? It seems normal to me that some types doesn't have related entities

Reply:
True. I'll drop the `warn`.